utils/src/resource_location.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "utils/resource_location.h"
+
+#include <filesystem>
+#include <regex>
+
+#include "network/uri/uri.hpp"
+
+#include "utils/path.h"
+#include "utils/platform.h"
+
+namespace hlasm_plugin::utils::path {
+namespace {
+const std::string untitled = "untitled";
+const std::string hlasm = "hlasm";
+} // namespace
+
+std::string uri_to_path(const std::string& uri)
+{
+    try
+    {
+        network::uri u(uri);
+
+        // vscode sometimes sends us uri in form 'untitled:Untitled-1',
+        // when user opens new file that is not saved to disk yet
+        if (!u.scheme().compare(untitled))
+            return uri;
+        // There is a hlasm specific schema - don't convert to path
+        if (!u.scheme().compare(hlasm))
+            return uri;
```

Comment:
I don't thing this should be here anymore, the function should perhaps not be called in the first place?